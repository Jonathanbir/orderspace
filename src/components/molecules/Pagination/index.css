.wrapper {
	display: flex;
	align-items: center;
	justify-content: center;

	& .page-wrapper {
		display: none;
		border: 1px solid var(--light-gray);
		background-color: white;
		width: 32px;
		height: 32px;
		border-radius: 8px;
		font-family: Montserrat, sans-serif;
		font-size: 16px;
		font-weight: 500;
		font-stretch: normal;
		font-style: normal;
		line-height: 32px;
		letter-spacing: normal;
		text-align: center;
		color: var(--light-gray);
		cursor: pointer;
		margin: 0 6px;
	}

	& .dots {
		margin: 0 14px;
	}

	& .active-page {
		border-color: var(--primary);
		color: white;
		background-color: var(--primary);
	}

	& .show-page {
		display: block;
	}

	& > svg {
		cursor: pointer;
		margin: 0 8px;

		&:hover > path {
			fill: #1faa39;
		}
	}
}

.pagination {
	margin-bottom: 20px;
}
