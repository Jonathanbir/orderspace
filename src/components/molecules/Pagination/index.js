import React from 'react';
import classnames from 'classnames';

import Previous from 'images/icon/prev.inline.svg';
import Next from 'images/icon/next.inline.svg';
import Dots from 'images/icon/atom-icon-action-ellipsis.inline.svg';

import styles from './index.css';

const Pagination = ({ className, activePage = 1, pages = 1, onClick = () => {} }) => {
	const pageArr = [];
	for (let i = 1; i <= pages; i += 1) {
		pageArr.push(i);
	}

	if (pageArr.length > 5) {
		return (
			<div className={classnames(styles.wrapper, className)}>
				<Previous onClick={() => activePage > pageArr[0] && onClick(activePage - 1)} />
				{pageArr.map((p, idx) => (
					<div style={{ display: 'flex', alignItems: 'center' }} key={p}>
						<div
							className={classnames(styles.pageWrapper, {
								[styles.activePage]: activePage === p,
								[styles.showPage]:
									idx === 0 ||
									idx === pageArr.indexOf(activePage) - 1 ||
									idx === pageArr.indexOf(activePage) ||
									idx === pageArr.indexOf(activePage) + 1 ||
									idx === pageArr.length - 1,
							})}
							onClick={() => onClick(p)}
							onKeyPress={() => {}}
							role="button"
							tabIndex={0}
						>
							{p}
						</div>
						{pageArr.indexOf(activePage) !== 0 &&
							pageArr.indexOf(activePage) !== 1 &&
							pageArr.indexOf(activePage) !== 2 &&
							idx === 0 && <Dots className={styles.dots} />}
						{pageArr.indexOf(activePage) !== pageArr.length - 1 &&
							pageArr.indexOf(activePage) !== pageArr.length - 2 &&
							pageArr.indexOf(activePage) !== pageArr.length - 3 &&
							idx === pageArr.length - 2 && <Dots className={styles.dots} />}
					</div>
				))}
				<Next onClick={() => activePage < pageArr[pageArr.length - 1] && onClick(activePage + 1)} />
			</div>
		);
	}

	return (
		<div className={classnames(styles.wrapper, className)}>
			<Previous onClick={() => activePage > pageArr[0] && onClick(activePage - 1)} />
			{pageArr.map(p => (
				<div
					key={p}
					className={classnames(styles.pageWrapper, {
						[styles.activePage]: activePage === p,
						[styles.showPage]: true,
					})}
					onClick={() => onClick(p)}
					onKeyPress={() => {}}
					role="button"
					tabIndex={0}
				>
					{p}
				</div>
			))}
			<Next onClick={() => activePage < pageArr[pageArr.length - 1] && onClick(activePage + 1)} />
		</div>
	);
};

export default Pagination;
