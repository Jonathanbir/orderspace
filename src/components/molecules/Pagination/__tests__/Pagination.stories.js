import React from 'react';
import { storiesOf } from '@storybook/react';

import Pagination from 'components/molecules/Pagination';

const stories = storiesOf('shop|molecules/Pagination', module);

stories.add('__interactive', () => <Pagination activePage={1} pages={5} />, {
	jest: 'Pagination',
});
