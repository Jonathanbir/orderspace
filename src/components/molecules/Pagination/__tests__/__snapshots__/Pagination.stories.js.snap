// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__interactive 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <test-file-stub />
    <div
      class="pageWrapper activePage showPage"
      role="button"
      tabindex="0"
    >
      1
    </div>
    <div
      class="pageWrapper showPage"
      role="button"
      tabindex="0"
    >
      2
    </div>
    <div
      class="pageWrapper showPage"
      role="button"
      tabindex="0"
    >
      3
    </div>
    <div
      class="pageWrapper showPage"
      role="button"
      tabindex="0"
    >
      4
    </div>
    <div
      class="pageWrapper showPage"
      role="button"
      tabindex="0"
    >
      5
    </div>
    <test-file-stub />
  </div>
</DocumentFragment>
`;
